server/redeemer.go

```diff
@@ -0,0 +1,303 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"math/big"
+	gonet "net"
+	"net/url"
+	"sync"
+	"time"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/golang/glog"
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/livepeer/go-livepeer/eth"
+	"github.com/livepeer/go-livepeer/net"
+	"github.com/livepeer/go-livepeer/pm"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+)
+
+const rpcTimeout = 8 * time.Second
+
+type redeemer struct {
+	recipient   ethcommon.Address
+	subs        sync.Map
+	eth         eth.LivepeerEthClient
+	sm          pm.SenderMonitor
+	quit        chan struct{}
+	liveSenders sync.Map
```

Comment:
How to clean this up ? 
No way to feasibly know when the queue for a sender is feasibly empty 

Why do we have this mapping in the first place?
`QueueTicket` fires a watchdog, but these are per sender and not per ticket. We'd want to avoid multiple watchdogs consuming the same channel to perform the same operation. The first to read from the channel will send and update to the subscribed clients but this spins up a lot of go routines until a ticket is redeemed. Therefore if a sender already exists in `liveSenders`  we don't spin up an additional watchdog and clean up the goroutine.